streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/StickyTaskAssignor.java

```diff
@@ -54,7 +60,7 @@ public void assign(final int numStandbyReplicas) {
     }
 
     private void assignStandby(final int numStandbyReplicas) {
-        for (final TaskId taskId : taskIds) {
+        for (final TaskId taskId : standbyTaskIds) {
```

Comment:
This is the main change; only loop over the stateful and changelog-enabled tasks when assigning standbys